src/mongocxx/write_concern.hpp

```diff
@@ -183,7 +204,14 @@ class MONGOCXX_API write_concern {
     ///
     /// @return The number of required nodes.
     ///
-    std::int32_t nodes() const;
+    stdx::optional<std::int32_t> nodes() const;
```

Comment:
Ok, got it.  Is it necessarily true that either nodes or acknowledge_level will be set?  Might they both be unset?